include/nil/crypto3/zk/snark/arithmetization/plonk/padding.hpp

```diff
@@ -76,6 +79,68 @@ namespace nil {
                     return padded_rows_amount;
                 }
 
+
+                template<typename FieldType, typename ArithmetizationParams, typename ColumnType>
+                std::uint32_t zk_padding(
+                    plonk_table<FieldType, ArithmetizationParams, ColumnType> &table,
+                    typename nil::crypto3::random::algebraic_engine<FieldType> alg_rnd = nil::crypto3::random::algebraic_engine<FieldType>()
+                ) {
+                    std::uint32_t usable_rows_amount = table.rows_amount();
+
+                    std::uint32_t padded_rows_amount = std::pow(2, std::ceil(std::log2(usable_rows_amount)));
+                    if (padded_rows_amount == usable_rows_amount)
+                        padded_rows_amount *= 2;
+
+                    if (padded_rows_amount < 8)
+                        padded_rows_amount = 8;
+
+                    //std::cout << "usable_rows_amount = " << usable_rows_amount << std::endl;
+                    //std::cout << "padded_rows_amount = " << padded_rows_amount << std::endl;
+
+                    for (std::uint32_t w_index = 0; w_index < table._private_table.witnesses_amount(); w_index++) {
+                        table._private_table._witnesses[w_index].resize(usable_rows_amount, FieldType::value_type::zero());
+                    }
+
+                    for (std::uint32_t pi_index = 0; pi_index < table._public_table.public_inputs_amount(); pi_index++) {
+                        table._public_table._public_inputs[pi_index].resize(usable_rows_amount, FieldType::value_type::zero());
+                    }
+
+                    for (std::uint32_t c_index = 0; c_index < table._public_table.constants_amount(); c_index++) {
+                        table._public_table._constants[c_index].resize(usable_rows_amount, FieldType::value_type::zero());
+                    }
+
+                    for (std::uint32_t s_index = 0; s_index < table._public_table.selectors_amount(); s_index++) {
+                        table._public_table._selectors[s_index].resize(usable_rows_amount, FieldType::value_type::zero());
+                    }
+
+
+                    for (std::uint32_t w_index = 0; w_index < table._private_table.witnesses_amount(); w_index++) {
+                        table._private_table._witnesses[w_index].resize(padded_rows_amount);
```

Comment:
Create some function, so this few lines will become:

table._private_table._witnesses = create_intialized_table(table._private_table.witnesses_amount(), usable_rows_amount, padded_rows_amount);

All the code will become 4 calls to this function.